dashboard/cron.py

```diff
@@ -226,24 +230,25 @@ def update_with_bq_access(self):
                     for table in tables:
                         if ("enriched_events" == table.table_id):
                             logger.debug('\t{}'.format("found table"))
-                            # loop through multiple course ids
-                            for UDW_course_id in Course.objects.get_supported_courses():
+                            # loop through multiple course ids, 10 at a time
+                            for udw_course_ids in split_list(Course.objects.get_supported_courses(), settings.CRON_BQ_IN_LIMIT):
 
                                 # query to retrieve all file access events for one course
-                                query = '''select CAST(SUBSTR(JSON_EXTRACT_SCALAR(event, "$.object.id"), 35) AS STRING) AS file_id,
-                                        SUBSTR(JSON_EXTRACT_SCALAR(event, "$.membership.member.id"), 29) AS user_id,
+                                query = """select CAST(SUBSTR(JSON_EXTRACT_SCALAR(event, '$.object.id'), 35) AS STRING) AS file_id,
+                                        SUBSTR(JSON_EXTRACT_SCALAR(event, '$.membership.member.id'), 29) AS user_id,
                                         datetime(EVENT_TIME) as access_time
                                         FROM event_store.events
-                                        where JSON_EXTRACT_SCALAR(event, "$.edApp.id") = \'http://umich.instructure.com/\'
-                                        and type = \'NavigationEvent\'
-                                        and JSON_EXTRACT_SCALAR(event, "$.object.name") = \'attachment\'
-                                        and JSON_EXTRACT_SCALAR(event, "$.action") = \'NavigatedTo\'
-                                        and JSON_EXTRACT_SCALAR(event, "$.membership.member.id") is not null
-                                        and SUBSTR(JSON_EXTRACT_SCALAR(event, "$.group.id"),31) = @course_id
-                                        '''
+                                        where JSON_EXTRACT_SCALAR(event, '$.edApp.id') = 'http://umich.instructure.com/'
+                                        and type = 'NavigationEvent'
+                                        and JSON_EXTRACT_SCALAR(event, '$.object.name') = 'attachment'
+                                        and JSON_EXTRACT_SCALAR(event, '$.action') = 'NavigatedTo'
+                                        and JSON_EXTRACT_SCALAR(event, '$.membership.member.id') is not null
+                                        and SUBSTR(JSON_EXTRACT_SCALAR(event, "$.group.id"),31) IN UNNEST(@course_ids)
+                                        """
                                 logger.debug(query)
+                                logger.debug(udw_course_ids)
                                 query_params =[
```

Comment:
Maybe it would be better to move the param (@course_ids) definition in front of the query line (line 237)

Reply:
This is just to debug print both statements, it doesn't use it until line 251. This debug statement could probably be improved.